src/components/MapController/MapController.js

```diff
@@ -19,51 +38,98 @@ var MapController = React.createClass({
   propTypes: {
     variable: React.PropTypes.string,
     meta: React.PropTypes.array,
+    comparand: React.PropTypes.string,
+    comparandMeta: React.PropTypes.array,
     onSetArea: React.PropTypes.func.isRequired,
   },
 
   mixins: [ModalMixin],
 
   /*
-   * State items also set from meta object array Includes: - dataset - wmstime -
-   * variable
+   * State items also set from meta object array 
+   * Includes:
+   *  - dataset
+   *  - wmstime
+   *  - variable
    */
   getInitialState: function () {
     return {
-      styles: 'boxfill/ferret',
-      timeidx: 0,
-      logscale: false,
+      scalarLogscale: false,
+      numberOfContours: 10,
+      contourLogscale: false,
     };
   },
-
+  
+  //this function handles user selection of all the straightforward
+  //parameters like logscale or palette.
   updateSelection: function (param, selection) {
     var update = {}; update[param] = selection;
     this.setState(update);
   },
 
+  //this function updates the timestamp displayed on the map and triggers 
+  //a rerender.
+  //timeidx is a stringified object with a resolution
+  //(monthly, annual, seasonal) and an index (0-11). For example
+  //{timeres: monthly, timeidx: 3} would represent April. 
+  //It's stringified because Selector won't pass an object.
   updateTime: function (timeidx) {
     this.setState({
       timeidx: timeidx,
       wmstime: this.state.times[timeidx],
     });
   },
 
-  updateDataset: function (uniqueId) {
-    // Updates dataset in state. Updates time value to match new dataset
-
-    this.selectedDataset = this.props.meta.filter(function (el) {
-      return el.unique_id === uniqueId;
-    })[0];
-
-    this.requestTimeMetadata(uniqueId).then(response => {
-      this.selectedDataset.times = response.data[uniqueId].times;
+  //this function stores a dataset selected by the user to state, 
+  //triggering a rerender to show the new dataset.
```

Comment:
Similar comment to just above. It's a bit clearer here, but documenting the likely effects of an update in a pub-sub system is perilous, and establishes, in comments, a coupling between parts that pub-sub by design decouples.